internal/auth/oidc/rewrapping.go

```diff
@@ -0,0 +1,54 @@
+package oidc
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+)
+
+func init() {
+	kms.RegisterTableRewrapFn(defaultAuthMethodTableName, authMethodRewrapFn)
+}
+
+func authMethodRewrapFn(ctx context.Context, dataKeyVersionId, scopeId string, reader db.Reader, writer db.Writer, kmsRepo *kms.Kms) error {
+	const op = "oidc.authMethodRewrapFn"
+	if dataKeyVersionId == "" {
+		return errors.New(ctx, errors.InvalidParameter, op, "missing data key version id")
+	}
+	if scopeId == "" {
+		return errors.New(ctx, errors.InvalidParameter, op, "missing scope id")
+	}
+	if reader == nil {
+		return errors.New(ctx, errors.InvalidParameter, op, "missing database reader")
+	}
+	if writer == nil {
+		return errors.New(ctx, errors.InvalidParameter, op, "missing database writer")
+	}
+	if kmsRepo == nil {
+		return errors.New(ctx, errors.InvalidParameter, op, "missing kms repository")
+	}
+	var authMethods []*AuthMethod
+	// There are indexes on (scope id, <other>), so we can query on scope and refine via key id.
+	// This is the fastest query we can use without creating a new index on key_id.
+	if err := reader.SearchWhere(ctx, &authMethods, "scope_id=? and key_id=?", []interface{}{scopeId, dataKeyVersionId}, db.WithLimit(-1)); err != nil {
+		return errors.Wrap(ctx, err, op, errors.WithMsg("failed to query sql for rows that need rewrapping"))
```

Comment:
The only tests we have are those using a real database, where errors are hard to induce. Personally I don't think there's much value in writing tests to exercise this sort of thing, as the test oracle is just your own test code, which can only test errors that you _expect_ to happen. If you want we can add some sqlmock tests. What do you think?

Reply:
I've come around to sqlmock tests for non-happy path unit tests but in this case that's not even necessary since you could just pass-in a db without seeding the appropriate data. 

you could test the kms unhappy paths with similar strategies or a simple mock.